core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java

```diff
@@ -15,67 +15,32 @@
  */
 package com.datastax.oss.driver.internal.core.os;
 
-import com.datastax.oss.driver.internal.core.util.Reflection;
-import java.lang.reflect.Method;
-import jnr.ffi.LibraryLoader;
 import jnr.ffi.Platform;
-import jnr.ffi.Pointer;
-import jnr.ffi.Runtime;
-import jnr.ffi.Struct;
-import jnr.ffi.annotations.Out;
-import jnr.ffi.annotations.Transient;
-import jnr.posix.POSIXFactory;
-import jnr.posix.util.DefaultPOSIXHandler;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /** A gateway to perform system calls. */
 public class Native {
 
-  private static final Logger LOG = LoggerFactory.getLogger(Native.class);
+  private static NativeImpl impl = new JnrNativeImpl();
```

Comment:
Suggestion: make `Native` the interface directly, in order to avoid the additional abstraction.
```java
public interface Native {
  static Native INSTANCE = new JnrNative();
  boolean isCurrentTimeMicrosAvailable();
  ...
}
public class JnrNative implements Native {
  @Override
  public boolean isCurrentTimeMicrosAvailable() { ... }
  ...
}
```
This will just require a small change to every call site: `Native.currentTimeMicros()` => `Native.INSTANCE.currentTimeMicros()`. But again this is internal code so we're allowed to do it, and it's cleaner.

Reply:
In isolation I'd probably agree with you, but keep in mind we'll be adding a different NativeImpl for Graal in JAVA-2663.  With that in mind I'd argue it's cleaner to maintain Native as a top-level facade and swap the composed class (the NativeImpl implementation) based on Graal/not-Graal.

Reply:
As far as adding another implementation, this doesn't change anything:
```java
public class GraalNative implements Native {
  ...
}
```
The `Native.impl` or `Native.INSTANCE` singleton is held in a static variable in both cases, so it shouldn't be any harder to initialize it based on some environment condition.

Reply:
This approach is isomorphic onto the static methods of Native in use now.  I'd argue this change doesn't really buy you much; you remove a layer of abstraction but (hopefully) the rename of NativeImpl to Libc makes the relationship between the two layers much clearer anyway.  Given that and a desire to minimize further churn on this PR I'd argue for keeping things as they are for now.